internal/oci/container.go

```diff
@@ -344,6 +344,24 @@ func (c *Container) exitFilePath() string {
 	return filepath.Join(c.dir, "exit")
 }
 
+// IsRunning is a function that checks if a specific pid is running
+// it is used to check a container state when we don't want (or don't trust) a `$runtime state` call
+func (c *Container) IsRunning() bool {
+	pid := c.state.Pid
+	process, err := os.FindProcess(pid)
```

Comment:
Is this necessary? We could directly make the syscall since we already know the pid.

Reply:
I suppose not, though I personally would rather use the standard lib instead of directly using syscall. I am worried about having to handle things that the lib takes care of (this isn't a great example, but for go 1.14, longer syscalls need eintr handling https://golang.org/doc/go1.14#runtime). My inclination is to leave as is

Reply:
`os.FindProcess(pid)` doesn't add any extra cost on unix as it just populates a struct with the specified PID.

Reply:
https://github.com/golang/go/blob/cb11c981df7b4dc40550ab71cc097c25d24d7a71/src/os/exec_unix.go#L97 :+1: